src/test/scala/chiselTests/RawModuleSpec.scala

```diff
@@ -228,4 +228,43 @@ class RawModuleSpec extends ChiselFlatSpec with Utils with FileCheck {
          |""".stripMargin
     )
   }
+
+  "RawModule marked as formal test" should "emit a formal test declaration" in {
+    class Foo extends RawModule {
+      FormalTest(this)
+      FormalTest(this, MapTestParam(Map("hello" -> StringTestParam("world"))))
+      FormalTest(
+        this,
+        MapTestParam(
+          Map(
+            "a_int" -> IntTestParam(42),
+            "b_double" -> DoubleTestParam(13.37),
+            "c_string" -> StringTestParam("hello"),
+            "d_array" -> ArrayTestParam(Seq(IntTestParam(42), StringTestParam("hello"))),
+            "e_map" -> MapTestParam(
+              Map(
+                "x" -> IntTestParam(42),
+                "y" -> StringTestParam("hello")
+              )
+            )
+          )
+        ),
+        "thisBetterWork"
+      )
+    }
+
+    generateFirrtlAndFileCheck(new Foo)(
+      """|CHECK-LABEL: formal Foo of Foo_2 :
```

Comment:
nit: maybe add a check for `module Foo_2` so we know that is indeed the module name? Could maybe use the capture / substitution functionality as well, so we could do something like `CHECK: module [[NAME:Foo.+]]` and later `CHECK: formal Foo of [[NAME]]`?

Reply:
Ah yeah, nice catch. Forgot that this actually runs through FileCheck with all the bells and whistles.

Reply:
Done 👍